ai_eval/multiagent.py

```diff
@@ -0,0 +1,651 @@
+"""Multi-agent AI XBlock."""
+
+import itertools
+import re
+import textwrap
+import typing
+
+import jinja2
+import pydantic
+from django.utils.translation import gettext_noop as _
+from jinja2.sandbox import SandboxedEnvironment
+from xblock.core import XBlock
+from xblock.exceptions import JsonHandlerError
+from xblock.fields import Boolean, Dict, List, Scope, String
+from xblock.validation import ValidationMessage
+from web_fragments.fragment import Fragment
+
+from .base import AIEvalXBlock
+from .llm import SupportedModels
+
+
+DEFAULT_SUPERVISOR_PROMPT = textwrap.dedent("""
+    You are a supervisor managing an interaction between the following agents: Coach, Character.
```

Comment:
@ArturGaspar Was this prompt provided by WGU?

Reply:
@pkulkark Yes.